### PR TITLE
Add sasmodels recipe

### PR DESCRIPTION
**Pull request by tschoonj**

### CODE REVIEW

File: sasmodels/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+test:
+  imports:
+    - sasmodels
+    - sasmodels.custom
+    - sasmodels.models
+
+about:
+  home: http://www.sasview.org
+  license: BSD
```
Comment (hoechenberger) [inline sasmodels/meta.yaml]: The license is `BSD-3-Clause`.
Please also consider adding the `license_family` (which would then be `BSD`)

You also _must_ add a `license_file` entry please.

File: sasmodels/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+requirements:
+  host:
+    - pip
+    - python
+  run:
+    - python
+    - numpy
+    - scipy
+    - bumps
+    - pyopencl
```
Comment (hoechenberger) [inline sasmodels/meta.yaml]: It seems that `bump` and `pyopencl` are only required if some [`extras`](https://github.com/SasView/sasmodels/blob/8803a3853d8388dbec0845dd84f394341fb00041/setup.py#L69) are installed, which you are currently not doing. Please decide what you want :)

Also, there seems to be a [Windows-only dependency](https://github.com/SasView/sasmodels/blob/8803a3853d8388dbec0845dd84f394341fb00041/setup.py#L34) that needs to be added.

File: sasmodels/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+test:
+  imports:
+    - sasmodels
+    - sasmodels.custom
+    - sasmodels.models
+
+about:
+  home: http://www.sasview.org
+  license: BSD
+  summary: Package for calculation of small angle scattering models using OpenCL
```
Comment (hoechenberger) [inline sasmodels/meta.yaml]: Could you please also add a `description` here?

Comment (beckermr) [reply sasmodels/meta.yaml]: A description is not absolutely needed.

File: sasmodels/meta.yaml (hunk 1 above)
Comment (tschoonj) [reply sasmodels/meta.yaml]: Done

File: sasmodels/meta.yaml (hunk 2 above)
Comment (tschoonj) [reply sasmodels/meta.yaml]: > It seems that bump and pyopencl are only required if some extras are installed, which you are currently not doing. Please decide what you want :)

Does it matter that I am not modifying the `build.script` command to add the extras? Pip won't be used to install the extra packages anyway right?

Comment (tschoonj) [reply sasmodels/meta.yaml]: The `tinycc` windows dependency is not installable via conda at the moment. It isn't necessary anyway if the system has OpenCL support.

**Comment by tschoonj** [comment]
If you have no further comments, would you mind merging this one in? Thanks!

**Comment by tschoonj** [comment]
Could someone look at this again? Thanks!

**Comment by chrisburr** [comment]
Sorry for the delay, it looks good to me so yes! 👍

**Comment by tschoonj** [comment]
Thanks!

**Comment by tschoonj** [comment]
It would appear that I accidentally placed the `sasmodels` folder at the repo root level instead of the recipes folder... Shall I open a PR to fix this?

**Comment by chrisburr** [comment]
Yes please and ping me in the description.

**Pull request merged**